inference/core/workflows/core_steps/models/foundation/florence_2.py

```diff
@@ -0,0 +1,233 @@
+)
+
+TASKS_WITH_PROMPT = [
+    "<CAPTION_TO_PHRASE_GROUNDING>",
+    "<REFERRING_EXPRESSION_SEGMENTATION>",
+    "<REGION_TO_SEGMENTATION>",
+    "<OPEN_VOCABULARY_DETECTION>",
+    "<REGION_TO_CATEGORY>" "<REGION_TO_DESCRIPTION>",
+]
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Florence-2 Model",
+            "short_description": "Run a multitask transformer model for a wide range of computer vision tasks.",
+            "long_description": "Florence-2 is a multitask transformer model that can be used for a wide range of computer vision tasks. It is based on the Vision Transformer architecture and has been trained on a large-scale dataset of images with a wide range of labels. The model is capable of performing tasks such as image classification, object detection, and image segmentation.",
+            "license": "MIT",
+            "block_type": "model",
+        }
+    )
+    type: Literal["Florence2Model", "Florence2"]
+    images: Union[WorkflowImageSelector, StepOutputImageSelector] = ImageInputField
+    model_id: Union[WorkflowParameterSelector(kind=[ROBOFLOW_MODEL_ID_KIND]), str] = (
+        Field(
+            title="Model",
+            default="florence-pretrains/1",
+            description="Roboflow model identifier",
+            examples=["florence-pretrains/1", "$inputs.model"],
+        )
+    )
+    vision_task: Union[
+        Literal[
+            "<OD>",
+            "<CAPTION_TO_PHRASE_GROUNDING>",
+            "<DENSE_REGION_CAPTION>",
+            "<REGION_PROPOSAL>",
+            "<OCR_WITH_REGION>",
+            "<REFERRING_EXPRESSION_SEGMENTATION>",
+            "<REGION_TO_SEGMENTATION>",
+            "<OPEN_VOCABULARY_DETECTION>",
+            "<REGION_TO_CATEGORY>",
+            "<REGION_TO_DESCRIPTION>",
+        ],
+        WorkflowParameterSelector(kind=[STRING_KIND]),
+    ] = Field(
+        description="The computer vision task to perform.",
+        default="<OPEN_VOCABULARY_DETECTION>",
+        examples=["<OPEN_VOCABULARY_DETECTION>"],
+    )
+    prompt: Union[WorkflowParameterSelector(kind=[LIST_OF_VALUES_KIND]), List[str]] = (
+        Field(
+            description="The accompanying prompt for the task (comma separated).",
+            examples=[["red apple", "blue soda can"], "$inputs.prompt"],
+        )
+    )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="parent_id", kind=[BATCH_OF_PARENT_ID_KIND]),
+            OutputDefinition(name="root_parent_id", kind=[BATCH_OF_PARENT_ID_KIND]),
+            OutputDefinition(name="image", kind=[BATCH_OF_IMAGE_METADATA_KIND]),
+            OutputDefinition(name="raw_output", kind=[BATCH_OF_DICTIONARY_KIND]),
+            OutputDefinition(name="structured_output", kind=[WILDCARD_KIND]),
```

Comment:
Prob need a special detections output to be able to nicely use it downstream (and make it an empty sv.Detections if not present)

Reply:
After reading the rest of the PR I think this is what `structured_output` is supposed to be -- why not call it `detections` (or do we use `predictions` as our convention elsewhere?) & give it a sv.Detections type

Reply:
structured_output is a paradigm Emily used on the CogVLM block that I inherited. For this model we could use predictions instead since they are all sv.Detections, though for the time being, I found that those sv detections were not compatible with many sv operations because of lack of data (no confidence or class id)

Reply:
Which operations? We def will want to visualize them, crop them, count them, use them as inputs to SAM, upload them to Roboflow as annotations, etc

We should document & fix those downstream issues vs avoiding using sv.Detections (which is our standard for grounded data coming from other models like YOLO-World as well)

Reply:
yeah of course use sv.Detections, I started looking at where to fix the issues. I get a variety of errors with sv annotator blocks without full detections. Currently confidence and class id are set to None which doesnt play well

Reply:
Maybe we should set them to 1 and 0?

Reply:
Or look at how YOLO-World handles Class ID for Open Set

Reply:
yolo world sets the ids to the index in the list of classes provided but that only works for open vocabulary object detection where the classes are provided, not for region text location, object detection, or other tasks where a list of ids isnt provided. I think we could just fake it for now and arbitrarily assign ids to each class but we could get issues where there are visual artifacts across frames for some future blocks that rely on ids